tsinfer/formats.py

```diff
@@ -1971,14 +2020,22 @@ def sites(self, ids=None):
         alleles = self.sites_alleles[:]
         metadata = self.sites_metadata[:]
         time = self.sites_time[:]
+        ancestral_allele_index = self.sites_ancestral_allele_index[:]
         if ids is None:
             ids = np.arange(0, self.num_sites, dtype=int)
         for j in ids:
+            aa = ancestral_allele_index[j]
```

Comment:
I don't think we should reorder the alleles here, let's just keep them in the order specified by the user.

Reply:
I'm worried that people will use this allele list to index into the genotypes array (it's confusing that you can get the alleles using `variant.site.alleles` and `variant.alleles`. Perhaps the site should return a `set()` of allelic states, like `ts.site(0).alleles` does? In this case I think we do want to return the actual value for the ancestral_allele, rather than the index, as you can't us it to index into a set.

Reply:
That's OK though, isnt it, as we won't reorder the genotypes unless specifically asked to (easy to update internal methods to do this)

Reply:
I thought we were setting `recode_ancestral=True` by default? But I can see that we might want *not* to do that by default, and set it explicitly to `True` in the tsinfer internal routines.

Reply:
Let's set recode_ancestral=False by default, probably no good reason to do this outside of tsinfer.

Reply:
Agree.